src/components/trade/TradesTable/index.tsx

```diff
@@ -0,0 +1,186 @@
+  getOrderExecutedPrice,
+  formatCalculatedPriceToDisplay,
+  formatExecutedPriceToDisplay,
+} from 'utils'
+import { HelpTooltip } from 'components/Tooltip'
+import StyledUserDetailsTable, {
+  StyledUserDetailsTableProps,
+  EmptyItemWrapper,
+} from '../../common/StyledUserDetailsTable'
+import Icon from 'components/Icon'
+import TradeOrderType from 'components/common/TradeOrderType'
+import { Surplus } from './Surplus'
+import { RAW_ORDER } from '../../../../test/data'
+
+const Wrapper = styled(StyledUserDetailsTable)`
+  > thead > tr,
+  > tbody > tr {
+    grid-template-columns: 10rem 4rem repeat(2, 10rem) repeat(2, 14rem) 10rem 10rem 1fr;
+  }
+`
+
+const TxHash = styled.div`
+  max-width: 10rem;
+  overflow: hidden;
+  text-overflow: ellipsis;
+`
+
+function isTokenErc20(token: TokenErc20 | null | undefined): token is TokenErc20 {
+  return (token as TokenErc20).address !== undefined
+}
+
+function formattedAmount(erc20: TokenErc20 | null | undefined, amount: BigNumber): string {
+  if (!isTokenErc20(erc20)) return '-'
+
+  return erc20.decimals ? formatSmartMaxPrecision(amount, erc20) : amount.toString(10)
+}
+
+function getLimitPrice(trade: Trade, isPriceInverted: boolean): string {
+  if (!trade.buyToken || !trade.sellToken) return '-'
+
+  const calculatedPrice = getOrderLimitPrice({
+    buyAmount: trade.buyAmount,
+    sellAmount: trade.sellAmount,
+    buyTokenDecimals: trade.buyToken.decimals,
+    sellTokenDecimals: trade.sellToken.decimals,
+    inverted: isPriceInverted,
+  })
+
+  return formatCalculatedPriceToDisplay(calculatedPrice, trade.buyToken, trade.sellToken, isPriceInverted)
+}
+
+function getExecutedPrice(trade: Trade, isPriceInverted: boolean): string {
+  if (!trade.buyToken || !trade.sellToken) return '-'
+
+  const order: RawOrder = {
+    ...RAW_ORDER,
+    executedBuyAmount: trade.executedBuyAmount?.toString() || '',
+    executedSellAmount: trade.executedSellAmount?.toString() || '',
+    executedFeeAmount: trade.executedFeeAmount?.toString() || '',
+  }
+
+  const calculatedPrice = getOrderExecutedPrice({
+    order,
+    buyTokenDecimals: trade.buyToken.decimals,
+    sellTokenDecimals: trade.sellToken.decimals,
+    inverted: isPriceInverted,
+  })
+
+  return formatExecutedPriceToDisplay(calculatedPrice, trade.buyToken, trade.sellToken, isPriceInverted)
+}
+
+const tooltip = {
+  tradeID: 'A unique identifier ID for this trade.',
+}
+
+export type Props = StyledUserDetailsTableProps & {
+  trades: Trade[]
+}
+
+interface RowProps {
+  trade: Trade
+  isPriceInverted: boolean
+}
+
+const RowOrder: React.FC<RowProps> = ({ trade, isPriceInverted }) => {
+  const { executionTime, buyToken, buyAmount, sellToken, sellAmount, kind, orderId } = trade
+
+  return (
+    <tr key={orderId}>
+      <td>
+        {
+          <RowWithCopyButton
+            className="span-copybtn-wrap"
+            textToCopy={orderId}
+            contentsToDisplay={<Link to={`/trades/${trade.orderId}`}>{orderId}</Link>}
```

Comment:
Point the link to the order details page, and display the shortened order id https://github.com/gnosis/gp-ui/blob/434997a0260c7e8fb663ba675c70bcba01166a81/src/utils/operator.ts#L228

```suggestion
            contentsToDisplay={<Link to={`/orders/${trade.orderId}`}>{getShortOrderId(orderId)}</Link>}
```